studio/pages/account/security.tsx

```diff
@@ -0,0 +1,246 @@
+import ShimmeringLoader from 'components/ui/ShimmeringLoader'
+import { useMfaListFactorsQuery } from 'data/profile/mfa-list-factors-query'
+import { useMfaUnenrollMutation } from 'data/profile/mfa-unenroll-mutation'
+import { useStore } from 'hooks'
+import { useState } from 'react'
+import { NextPageWithLayout } from 'types'
+import {
+  Accordion,
+  AlertDescription_Shadcn_,
+  AlertTitle_Shadcn_,
+  Alert_Shadcn_,
+  Badge,
+  Button,
+  IconAlertCircle,
+  IconAlertTriangle,
+  IconSmartphone,
+  IconTrash,
+  Modal,
+} from 'ui'
+
+const Security: NextPageWithLayout = () => {
+  const { data } = useMfaListFactorsQuery()
+
+  return (
+    <div className="1xl:px-28 mx-auto flex flex-col px-5 pt-6 pb-14 lg:px-16 xl:px-24 2xl:px-32">
+      <div className="flex items-center justify-between">
+        <FormHeader
+          title="Multi-factor authentication"
+          description="Add an additional layer of security to your account by requiring more than just a password to sign in."
+        />
+      </div>
+      <Accordion
+        openBehaviour="multiple"
+        chevronAlign="left"
+        type="bordered"
+        className="max-w-4xl"
+        defaultValue={['auth-app']}
+      >
+        <Accordion.Item
+          header={
+            <div className="flex flex-row justify-between items-center p-2 w-full">
+              <div className="flex flex-row gap-2 items-center p-2">
+                <IconSmartphone />
+                <span className="text-lg">Authenticator app</span>
+              </div>
+
+              {data ? (
+                <>
+                  {data.totp.length === 0 && <Badge color="red">0 apps configured</Badge>}
+                  {data.totp.length === 1 && <Badge color="yellow">1 apps configured</Badge>}
+                  {data.totp.length === 2 && <Badge color="green">2 apps configured</Badge>}
+                </>
+              ) : null}
+            </div>
+          }
+          id="auth-app"
+          className="flex flex-row gap-8"
+        >
+          <ProfileCard />
+        </Accordion.Item>
+      </Accordion>
+    </div>
+  )
+}
+
+Security.getLayout = (page) => (
+  <AccountLayout title="Security" breadcrumbs={[{ key: 'security', label: 'Security' }]}>
+    {page}
+  </AccountLayout>
+)
+
+export default Security
+
+const ProfileCard = () => {
+  const [isAddNewFactorOpen, setIsAddNewFactorOpen] = useState(false)
+  const [factorToBeDeleted, setFactorToBeDeleted] = useState<string | null>(null)
+  const { data, isLoading, isError, isSuccess, error } = useMfaListFactorsQuery()
+
+  return (
+    <>
+      <section className="px-6 pt-4">
+        <div>
+          <span className="text-sm prose color-slate-1000">
+            Generate one-time passwords via authenticator apps like 1Password, Authy, etc. as a
+            second factor to verify your identity when prompted during sign-in.
+          </span>
+        </div>
+        {isLoading && (
+          <div className="py-2">
```

Comment:
<GenericSkeletonLoader />

Reply:
also is the py-2 required? otherwise lets just wrap isLoading, isError, isSuccess with the py-n instead

Reply:
mainly just to keep things consistent cause below we've got py-4 for the error state